utils/paramfetch/src/lib.rs

```diff
@@ -48,18 +48,28 @@ struct ParameterData {
 }
 
 #[inline]
-fn param_dir() -> String {
-    std::env::var(DIR_ENV).unwrap_or_else(|_| PARAM_DIR.to_owned())
+fn param_dir(mut data_dir: PathBuf) -> PathBuf {
+    if let Ok(path_buf_string) = std::env::var(PathBuf::from(DIR_ENV)) {
+        PathBuf::from(path_buf_string)
+    } else {
+        data_dir.push(PARAM_DIR);
+        data_dir
+    }
+}
+
+pub fn set_dir_env(data_dir: PathBuf) {
+    std::env::set_var(DIR_ENV, data_dir.clone());
 }
 
 /// Get proofs parameters and all verification keys for a given sector size given
 /// a param JSON manifest.
 pub async fn get_params(
+    data_dir: PathBuf,
```

Comment:
`data_dir` is immutable so we don't need to own it. Try with a `&Path` instead.

Reply:
Fixed